okhttp/src/main/java/io/grpc/okhttp/OkHttpClientTransport.java

```diff
@@ -476,6 +522,12 @@ public void shutdownNow(Status reason) {
         it.remove();
         entry.getValue().transportReportStatus(reason, false, new Metadata());
       }
+      if (keepAliveManager != null) {
+        // We don't have any active streams. No need to do keepalives any more.
+        // Again, we have to call this inside the lock to avoid the race between onTransportIdle
+        // and onTransportActive.
+        keepAliveManager.onTransportIdle();
```

Comment:
I see both maybeClearInUse and setInUse actually consider pending streams. IIUC, when there's no connection, all streams will be pending streams. We probably don't want to do keepalives when there're only pending streams.